sdk/core/System.TypeSpec/src/RequestOptions.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ComponentModel;
+using System.Diagnostics;
+using System.ServiceModel.Rest.Core;
+using System.Threading;
+
+namespace System.ServiceModel.Rest;
+
+public class RequestOptions : PipelineOptions
```

Comment:
Once RequestOptions inherits from PipelineOptions, and holds a pipeline, Azure.Core's Azure.RequestContext also holds a pipeline that Azure.Core-based clients won't use until we rewrite all of them; probably not a dependency we want to commit to.  Does that mean we want to separate the ClientModel PipelineOptions/RequestOptions story completely from the Azure.Core ClientOptions/RequestContext story?  In the sense that there will be no inheritance of RequestContext from RequestOptions?  Or did you have other thoughts here, @KrzysztofCwalina?

Reply:
If the cached pipeline is not public, we don't need to use it in Azure.Core. i.e. the fact that it's cached is just an optimization, isn't it?

Reply:
If we don't use it in Azure.Core, I think you're right that we don't have a problem with a frozen pipeline, but I think it means Azure.Core-based clients will have two pipelines in-memory, and there will be public API surface with multiple ways to set things like the RetryPolicy...